.github/workflows/publish-docker-tester.yml

```diff
@@ -97,7 +97,34 @@ jobs:
           else
             echo "DOCKER_IMAGE_SHA=${{ github.sha }}" >> $GITHUB_ENV
           fi
+      # dpservice-bin tests
       - name: Run tests
         id: run_tests
         run: |
           docker run --rm --privileged --mount type=bind,source=/dev/hugepages,target=/dev/hugepages ghcr.io/ironcore-dev/dpservice-tester:${{ env.DOCKER_IMAGE_SHA }}
+      - name: Docker cleanup
+        if: always()
+        run: |
+          for i in $(docker ps -q); do docker stop $i; done
+      # dpservice-go tests
+      - name: Run checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
We need to use `${{ env.DOCKER_IMAGE_SHA }}` as ref here. otherwise it will not check out ok when we build for the main branch. (when the trigger of the action is not "pull_request")
@vlorinc @PlagueCZ

Reply:
It's fixed now and tests were successful.